docs/spec/relayer/Relayer.tla

```diff
@@ -5,111 +5,31 @@
  It instantiates the module Environment, which takes care of the chain logic. 
  ***************************************************************************)
 
-EXTENDS Naturals, FiniteSets
+EXTENDS Naturals, FiniteSets, RelayerDefinitions
 
-CONSTANTS MaxHeight \* maximal height of all the chains in the system 
+CONSTANTS GenerateClientDatagrams, \* toggle generation of client datagrams
+          GenerateConnectionDatagrams, \* toggle generation of connection datagrams
+          GenerateChannelDatagrams \* toggle generation of channel datagrams
 
-VARIABLES chains, \* a function that assigns a chain record to each chainID 
+ASSUME /\ GenerateClientDatagrams \in BOOLEAN 
+       /\ GenerateConnectionDatagrams \in BOOLEAN 
+       /\ GenerateChannelDatagrams \in BOOLEAN 
+
+CONSTANTS Heights \* set of possible heights of the chains in the system 
+
+VARIABLES chainAstore, \* store of ChainA
+          chainBstore, \* store of ChainB
           outgoingDatagrams, \* a function that assigns a set of pending datagrams 
                              \* outgoing from the relayer to each chainID 
-          relayerChainHeights, \* a function that assigns a height to each chainID         
-          turn  
+          relayerHeights \* a function that assigns a height to each chainID         
```

Comment:
Relayer in principle does not maintain latest height (header) for each chain. It actually needs to send client update for the corresponding datagrams. For this it needs to be able to create clientDatagrams and for this is needs access to trusted headers. This is what light client will provide. We should think how we want to model this, but for now I would suggest to remove relayerHeights as it does not really add anything useful to the model as we create client updates based on src chain height and src client height on the destination chain.